packages/e2e-test-utils-playwright/src/admin/visit-site-editor.ts

```diff
@@ -37,31 +37,32 @@ export async function visitSiteEditor(
 		query.set( 'canvas', canvas );
 	}
 
-	const canvasLoader = this.page.locator(
-		// Spinner was used instead of the progress bar in an earlier version of
-		// the site editor.
-		'.edit-site-canvas-loader, .edit-site-canvas-spinner'
-	);
-
 	await this.visitAdminPage( 'site-editor.php', query.toString() );
 
-	// Try waiting for the canvas loader to appear first, so that the locator
-	// that waits for it to disappear doesn't resolve prematurely.
-	await canvasLoader.waitFor().catch( () => {} );
-
 	/**
 	 * @todo This is a workaround for the fact that the editor canvas is seen as
 	 * ready and visible before the loading spinner is hidden. Ideally, the
 	 * content underneath the loading overlay should be marked inert until the
 	 * loading is done.
 	 */
-	await canvasLoader.waitFor( {
-		state: 'hidden',
-		// Bigger timeout is needed for larger entities, like the Large Post
-		// HTML fixture that we load for performance tests, which often doesn't
-		// make it under the default timeout value.
-		timeout: 60_000,
-	} );
+	if ( ! query.size || postId || canvas === 'edit' ) {
+		const canvasLoader = this.page.locator(
+			// Spinner was used instead of the progress bar in an earlier
+			// version of the site editor.
+			'.edit-site-canvas-loader, .edit-site-canvas-spinner'
+		);
+
+		// Wait for the canvas loader to appear first, so that the locator that
+		// waits for the hidden state doesn't resolve prematurely.
+		await canvasLoader.waitFor( { state: 'visible' } );
```

Comment:
Just wondering: why did it previously pass the check for the existence of the canvas loader, and then continue on to never disappear?

Reply:
It only passed the check because [there was an empty `catch`](https://github.com/WordPress/gutenberg/pull/61629#discussion_r1599825668) for the scenarios in which the loader disappeared before the test reached that line. It was a premature optimization on my end – we didn't need it, and it only made things worse by swallowing the error that would prevent the original PR from being merged. 😅